### DISCUSSION TITLE
How to enable the TOTP for a new user in a customed organization?

### DISCUSSION THREAD

**Discussion by SidneySun** [casdoor/casdoor#2100] [Q&A] [answered]
Can I enable the TOTP for a non-admin user?

I create a new organization EXAMPLE and a non-admin Alice. I wanna Alice to use the TOTP.
- If I login into the admin account, then edit the user Alice in the users, and set Alice's MFA as App. The admin's MFA will be TOTP but not Alice's.
- If I login into the admin account, then edit the organization EXAMPLE. The MFA is only one of Phone and Email.
- Alice isn't in built-in/app-built-in, so I can't log into the casdoor app to set the TOTP.
Can anyone help me? Thanks!

**Comment by SidneySun** [comment]
Could you help me? Thanks! @hsluoyz @leo220yuyaodog

**Reply by leo220yuyaodog** [reply]
About the question 2,
> If I login into the admin account, then edit the organization EXAMPLE. The MFA is only one of Phone and Email. 

You can add the MFA type 

![image](https://github.com/casdoor/casdoor/assets/71440988/b392db46-f178-4996-9b1a-dba01750da7e)

About question 1 and 3, now Casdoor just supports user to enbale MFA by itself.  You can set the rule required. User will be requied to enabled the MFA before login. See details https://casdoor.org/docs/organization/mfa-items

**Reply by SidneySun** [reply]
There are Email and Phone with the version v1.352.0.
![image](https://github.com/casdoor/casdoor/assets/41945071/6759af79-9452-47fa-b692-39075c60f7bd)
Is this expected?

**Reply by leo220yuyaodog** [reply, marked as answer]
Please update to at least v1.360 https://github.com/casdoor/casdoor/commit/347d3d2b53a1ef9e46f84661fad9dc3f8b99a0f5

**Reply by SidneySun** [reply]
I'm sorry that I didn't read the docs carefully. Thank you!

**Reply by SidneySun** [reply]
@leo220yuyaodog emm... I failed to set the TOTP. It will work if the user is in the default organization built-in.  But if the user is in the custom organization EXAMPLE, I will get the error _Failed to get the application_. 
![image](https://github.com/casdoor/casdoor/assets/41945071/d6f278e5-761a-4da8-bac8-7cbfef397ca1)

**Comment by leo220yuyaodog** [comment]
> the TOTP. It will work if the user is in the default organization built-in. But if t

Which version? I suggest that update to the latest version

**Reply by SidneySun** [reply]
It is the latest v1.367.0.
![image](https://github.com/casdoor/casdoor/assets/41945071/8076954c-887d-444d-928b-1b252ae2bce3)

**Comment by javi17** [comment]
how enable mfa by my self?
![Screenshot from 2024-09-05 16-16-41](https://github.com/user-attachments/assets/3a5eef91-b450-4778-9adb-42e21e07ed74)
